### PR TITLE
Add dependabot and renovate documentation page

### PR DESCRIPTION
**Pull request by zanieb**
cc @mkniewallner

### CODE REVIEW

File: docs/guides/integration/dependency-bots.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+to `uv.lock` files.
+
+```json5 title="renovate.json5"
+{
+  $schema: "https://docs.renovatebot.com/renovate-schema.json",
+  lockFileMaintenance: {
+    enabled: true,
+  },
+}
+```
```
Comment (mkniewallner) [inline docs/guides/integration/dependency-bots.md:17]: `lockFileMaintenance` is not required for uv support, it mostly allows refreshing transitive dependencies (or also direct dependencies if using range of versions for dependencies in `pyproject.toml`) in the lock file on a regular basis, so this should probably be a dedicated paragraph.

I'd probably go with something like:

```suggestion
Renovate will detect the presence of `uv.lock` file to determine that uv is used for managing
dependencies, and will suggest updates to
both [project dependencies](../../concepts/dependencies.md#project-dependencies), [optional dependencies](../../concepts/dependencies.md#optional-dependencies)
and [development dependencies](../../concepts/dependencies.md#development-dependencies), updating
`uv.lock` file along the way.

It is also possible to refresh `uv.lock` file on a regular basis (for instance to update transitive
dependencies) by
enabling [`lockFileMaintenance`](https://docs.renovatebot.com/configuration-options/#lockfilemaintenance)
option:

```json5 title="renovate.json5"
{
  $schema: "https://docs.renovatebot.com/renovate-schema.json",
  lockFileMaintenance: {
    enabled: true,
  },
}
```
```

**Review by mkniewallner — COMMENTED** [review]
Since support for Renovate was only recently added, and Renovate GitHub app does not yet use a version that added support for uv, do you mind waiting a bit before merging this one? I'd ideally prefer to test that it works correctly once the new version is used by Renovate GitHub app (which I'll make sure to do as soon as it's available, and report status here), to make sure we don't advertise for something that does not work :smile:

File: docs/guides/integration/dependency-bots.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+!!! note
+
+    `uv pip compile` outputs such as `requirements.txt` are not yet supported by Renovate.
```
Comment (mkniewallner) [inline docs/guides/integration/dependency-bots.md:21]: Created a dedicated GitHub discussion for that (https://github.com/renovatebot/renovate/discussions/30899) that will hopefully be converted to an issue, to better track the implementation. Will report back if it ends up being converted to an issue.

Comment (mkniewallner) [reply docs/guides/integration/dependency-bots.md:21]: An issue now exists: https://github.com/renovatebot/renovate/issues/30909.

File: docs/guides/integration/dependency-bots.md (hunk 1 above)
Comment (zanieb) [reply docs/guides/integration/dependency-bots.md:17]: Thanks!

File: docs/guides/integration/dependency-bots.md (hunk 2 above)
Comment (zanieb) [reply docs/guides/integration/dependency-bots.md:21]: Awesome, I'll link to that.

**Review by charliermarsh — APPROVED** [review]

**Comment by zanieb** [comment]
@mkniewallner just let me know when it's ready.

I made some edits as you suggested — let me know if there's anything incorrect.

File: docs/guides/integration/dependency-bots.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+uv is supported by [Renovate](https://github.com/renovatebot/renovate).
+
+Renovate uses the presence of a `uv.lock` file to determine that uv is used for managing
+dependencies, and will suggest upgrades to
+[project dependencies](../../concepts/dependencies.md#project-dependencies),
+[optional dependencies](../../concepts/dependencies.md#optional-dependencies) and
+[development dependencies](../../concepts/dependencies.md#development-dependencies). Renovate will
+update both the `pyproject.toml` and `uv.lock` files.
+
+The lockfile on also be refreshed on a regular basis (for instance to update transitive
```
Comment (mkniewallner) [inline docs/guides/integration/dependency-bots.md:14]:
```suggestion
The lockfile can also be refreshed on a regular basis (for instance to update transitive
```

File: docs/guides/integration/dependency-bots.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+```
+
+!!! note
+
+    `uv pip compile` outputs such as `requirements.txt` are not yet supported by Renovate.
+    Progress can be tracked at [renovatebot/renovate#30909](https://github.com/renovatebot/renovate/issues/30909).
+
+## Dependabot
+
+Support for uv  is not yet available. Progress can be tracked at
```
Comment (mkniewallner) [inline docs/guides/integration/dependency-bots.md:35]:
```suggestion
Support for uv is not yet available. Progress can be tracked at
```

**Comment by mkniewallner** [comment]
> @mkniewallner just let me know when it's ready.
> 
> I made some edits as you suggested — let me know if there's anything incorrect.

Unfortunately, Renovate GitHub app still uses an old version that doesn't have the changes made to support uv for now, so I wasn't able to test that it works as expected on projects I've switched to uv. But I'm still regularly checking :slightly_smiling_face:

**Comment by mkniewallner** [comment]
Renovate GitHub app has been updated, and it seems to work fine (https://github.com/mkniewallner/showcase-uv-renovate/pull/13, https://github.com/mkniewallner/showcase-uv-renovate/pull/15), so this PR should be ready to merge.

**Pull request merged**